src/views/Editor.vue

```diff
@@ -446,7 +447,29 @@ export default {
 				this.$store.commit('loadHints', this.selectedSchema.ui)
 				this.$store.commit('loadData', Object(data.dataset))
 				this.$store.commit('setMetadata', { id, schemaId: this.selectedSchema.id })
-				this.qvainData = data
+				this.qvainData = data				
+				const languages = {}
+				let Ajv = require('ajv')
+				const ajv = new Ajv({ allErrors: true })				
+				ajv.addKeyword('$deref', {
+					type: 'object',
+					compile: function (sch, parentSchema) {
+						return function (data) {
+							if (sch === "#/definitions/langString") {
+								for (const lang in data) {
+									languages[lang] = true
+								}
+							}
+						}
+					},
+					valid: true,
+					modifying: true,
+					errors: false,
+				})
+				const validate = ajv.compile(this.$store.state.schema)
+				validate(data.dataset)			
```

Comment:
I'd suggest creating a new function for these added lines and creating a new file (e.g. lib/ajv.js) for it. Then you import the function in this file.

Reply:
added code